### ISSUE TITLE
ForemTag for embed may be too optimistic for domain filtering

### ISSUE THREAD

**Issue by jeremyf** [forem/forem#16922]
In the following URLs, we're using the domain names (e.g. `dev.to`) as the literate regular expression.

https://github.com/forem/forem/blob/c6c30d58abbd1ae2d0d6f6049b1ca5b202d49a04/app/liquid_tags/forem_tag.rb#L2-L5

That is to say our regexpressions expand to:

```ruby
/https:\/\/dev.to\/?/
```

With the following quick test all of the results matches return `true`.
```ruby
REGEXP = /https:\/\/dev.to\/?/

puts REGEXP.match?("https://dev.to")
puts REGEXP.match?("https://devnto")
puts REGEXP.match?("https://devntomuch.magic")
puts REGEXP.match?("https://dev.totalrecall.com")
puts REGEXP.match?("https://devnto.com")
```

I do not believe this is a critical issue but does open up for potential oddities.

**Comment by amywtlin** [comment]
@jeremyf Agreed that it's not a blocker but a good improvement. This feels like a small win to me - do you agree though?

**Comment by msarit** [comment]
Note for myself:

```ruby
REGEXP = /https:\/\/dev\.to\/\b([\w-]+)?/
```
